format/spec.md

```diff
@@ -444,6 +449,9 @@ Sorting floating-point numbers should produce the following behavior: `-NaN` < `
 
 A data or delete file is associated with a sort order by the sort order's id within [a manifest](#manifests). Therefore, the table must declare all the sort orders for lookup. A table could also be configured with a default sort order id, indicating how the new data should be sorted by default. Writers should use this default sort order to sort the data on write, but are not required to if the default order is prohibitively expensive, as it would be for streaming writes.
 
+Note:
+
+1. `variant` columns are not valid for sorting.
```

Comment:
I thought whether a variant is orderable is determined by engines per pervious discussion. Are we explicitly saying that all variants are not orderable?

Reply:
@flyrain Yeah. The engines can support ordering actually. Here basically we will not define the sorting order. I'm wondering if we should define one in the future. 

So I will update to  "The ability to sort variant columns and the specific sort order is determined by the engines."

Reply:
We are only saying that Variant values cannot be present in an Iceberg sort order. Engines can sort if they choose.

We could also define a sort order if we wanted, but this seems like a good place to start.

Reply:
Do we need to explicitly say "Variant values cannot be present in an Iceberg sort order. " in the spec?

Reply:
I think we should specifically write that. Don't we do so for Maps and Arrays?

Reply:
I don't see we call out anything for Maps and Arrays in sorting section. But the code has the check to error out with "Cannot sort by non-primitive source field". 

I don't add explicitly for Variant (just like Maps and Arrays) either but let me know if that makes sense .